product-service/lib/lambda/createProduct/index.ts

```diff
@@ -0,0 +1,98 @@
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import { DynamoDBDocumentClient, TransactWriteCommand } from "@aws-sdk/lib-dynamodb";
+import { randomUUID } from "crypto";
+import { Product } from '../../../src/products';
+import { Stock } from '../../../src/stocks';
+
+const productsTableName = process.env.PRODUCTS_TABLE_NAME!;
+const stocksTableName = process.env.STOCKS_TABLE_NAME!;
+
+const client = new DynamoDBClient({region: "eu-west-1"});
+const db = DynamoDBDocumentClient.from(client);
+
+export const createProductHandler = async (event: any) => {
+  try {
+    console.log("Received event:", JSON.stringify(event, null, 2));
+    const { title, description, price, count } = JSON.parse(event.body);
+    console.log('Parsed product and stock data:', { title, description, price, count });
+
+    // Input validation for product data
+    if (!title || !description || price === undefined || price === null || isNaN(price) || price <= 0) {
```

Comment:
It might also be a good thing to add a length check for title and description, so we can't create products with empty strings.

Reply:
I will improve it.